instance/models/mixins/openedx_database.py

```diff
@@ -75,6 +119,100 @@ def mongo_database_names(self):
         """
         return [self.mongo_database_name, self.forum_database_name]
 
+    @property
+    def migrate_user(self):
+        """
+        Name of migration user
+        """
+        return self._get_mysql_user_name("migrate")
+
+    @property
+    def read_only_user(self):
+        """
+        Name of read_only user
+        """
+        return self._get_mysql_user_name("read_only")
+
+    @property
+    def admin_user(self):
+        """
+        Name of admin user
+        """
+        return self._get_mysql_user_name("admin")
+
+    @property
+    def global_users(self):
+        """
+        List of MySQL users with global privileges (i.e., privileges spanning multiple databases)
+        """
+        return self.migrate_user, self.read_only_user, self.admin_user
+
+    def _get_mysql_database_name(self, suffix):
+        """
+        Build unique name for MySQL database using suffix.
+
+        To generate a unique name, this method adds an underscore and the specified suffix
+        to the database_name of this instance.
+
+        database_name can be up to 50 characters long.
+        The maximum length for the name of a MySQL database is 64 characters.
+        To ensure that the generated name does not exceed that length,
+        suffix should not consist of more than 13 characters.
+        """
+        mysql_database_name = "{0}_{1}".format(self.mysql_database_name, suffix)
+        assert len(mysql_database_name) <= 64
+        return mysql_database_name
+
+    def _get_mysql_user_name(self, suffix):
+        """
+        Build unique name for MySQL user using suffix.
+
+        To generate a unique name, this method adds an underscore and the specified suffix
+        to the mysql_user of this instance.
+
+        mysql_user is 6 characters long.
+        To ensure that the generated name does not exceed that length,
+        suffix must not consist of more than 9 characters.
+        """
+        mysql_user_name = "{0}_{1}".format(self.mysql_user, suffix)
+        assert len(mysql_user_name) <= 16
+        return mysql_user_name
+
+    def _get_mysql_pass(self, user):
+        """
+        Build unique password for user, derived from MySQL password for this instance and user.
+        """
+        encoding = "utf-8"
+        key = bytes(source=self.mysql_pass, encoding=encoding)
+        msg = bytes(source=user, encoding=encoding)
+        return hmac.new(key, msg=msg, digestmod=hashlib.sha256).hexdigest()
+
+    def _get_database_suffix(self, name):
+        """
+        Return suffix that differentiates database identified by name from databases for other services.
+        """
+        return remove_prefix(self.mysql_database_name, name)
```

Comment:
Nit: I'm not sure it's worth extracting prefix removal to a separate `remove_prefix` function, it's only two lines and not reused anywhere.